sys/include/ztimer/xtimer_compat.h

```diff
@@ -71,7 +71,32 @@ static inline uint64_t xtimer_now_usec64(void)
 static inline void xtimer_sleep(uint32_t seconds)
 {
     /* TODO: use ZTIMER_SEC */
-    ztimer_sleep(ZTIMER_USEC, seconds * 1000000LU);
+    if (IS_ACTIVE(MODULE_ZTIMER_MSEC)) {
+        while (seconds > (UINT32_MAX / 1000LU)) {
+            ztimer_sleep(ZTIMER_MSEC, UINT32_MAX);
+            seconds -= UINT32_MAX / 1000LU;
+        }
```

Comment:
One could avoid the rounding issues by using

```suggestion
        const uint32_t max_seconds = UINT32_MAX / 1000LU;
        while (seconds > max_seconds) {
            ztimer_sleep(ZTIMER_MSEC, max_seconds * 1000);
            seconds -= max_seconds;
        }
```

Otherwise one would sleep 295 milliseconds (`UINT32_MAX % 1000`) too much per loop.

Reply:
good catch